bundles/org.openhab.core.crypto.cryptostore/src/main/java/org/openhab/core/crypto/cryptostore/CryptoStore.java

```diff
@@ -0,0 +1,328 @@
+    }
+
+    public KeyStore getKeyStore(String keystorePassword, byte[] keyString)
+            throws GeneralSecurityException, IOException, Exception {
+        KeyStore keystore = KeyStore.getInstance("JKS");
+        keystore.load(null, null);
+        byte[] pkcs8EncodedBytes = Base64.getDecoder().decode(getPrivKey(keyString));
+        PKCS8EncodedKeySpec keySpec = new PKCS8EncodedKeySpec(pkcs8EncodedBytes);
+        KeyFactory kf = KeyFactory.getInstance(this.keystoreAlgorithm);
+        keystore.setKeyEntry(this.alias, kf.generatePrivate(keySpec), keystorePassword.toCharArray(),
+                new java.security.cert.Certificate[] { getCert() });
+        if (!caCert.isEmpty()) {
+            String caAlias = this.caAlias;
+            keystore.setCertificateEntry(caAlias, getCaCert());
+        }
+        return keystore;
+    }
+
+    public void saveKeyStore(String keystorePassword, byte[] keyString)
+            throws GeneralSecurityException, IOException, Exception {
+        saveKeyStore(this.keystoreFileName, keystorePassword, keyString);
+    }
+
+    public void saveKeyStore(String keystoreFileName, String keystorePassword, byte[] keyString)
+            throws GeneralSecurityException, IOException, Exception {
+        FileOutputStream keystoreStream = new FileOutputStream(keystoreFileName);
+        KeyStore keystore = getKeyStore(keystorePassword, keyString);
+        keystore.store(keystoreStream, keystorePassword.toCharArray());
+    }
+
+    private X509Certificate generateSelfSignedCertificate(KeyPair keyPair, String distName)
+            throws GeneralSecurityException, OperatorCreationException {
+        final Instant now = Instant.now();
+        final Date notBefore = Date.from(now);
+        final Date notAfter = Date.from(now.plus(Duration.ofDays(365 * 10)));
+        X500Name name = new X500Name(distName);
+        X509v3CertificateBuilder certificateBuilder = new JcaX509v3CertificateBuilder(name,
+                BigInteger.valueOf(now.toEpochMilli()), notBefore, notAfter, name, keyPair.getPublic());
+        ContentSigner contentSigner = new JcaContentSignerBuilder("SHA256WithRSA").build(keyPair.getPrivate());
+        return new JcaX509CertificateConverter().setProvider(new BouncyCastleProvider())
+                .getCertificate(certificateBuilder.build(contentSigner));
+    }
+
+    public void generateNewKeyPair(byte[] keyString)
+            throws GeneralSecurityException, OperatorCreationException, IOException, Exception {
+        Key key = convertByteToKey(keyString);
+        KeyPairGenerator kpg = KeyPairGenerator.getInstance(this.keystoreAlgorithm);
+        kpg.initialize(this.keyLength);
+        KeyPair kp = kpg.generateKeyPair();
+        Security.addProvider(new BouncyCastleProvider());
+        Signature signer = Signature.getInstance("SHA256withRSA", "BC");
+        signer.initSign(kp.getPrivate());
+        signer.update("openhab".getBytes(StandardCharsets.UTF_8));
+        signer.sign();
```

Comment:
What's the purpose of this?

Reply:
That generates a self signed certificate.  That's the part of the function that actually signs the certificate.

Reply:
From what I understand of the JavaDoc, this only creates a signature for the string "openhab", and the returned signature gets discarded anyway.

Reply:
I think lines 308-310 is what signs the certificate.

Reply:
Yes, looking back this looks like an artifact I missed from a previous revision.  I can remove it.